lib/elixir/lib/calendar.ex

```diff
@@ -136,6 +136,26 @@ defmodule Calendar do
   """
   @callback day_of_week(year, month, day) :: non_neg_integer()
 
+  @doc """
+  Calculates the day of the year from the given `year`, `month`, and `day`.
+  """
+  @callback day_of_year(year, month, day) :: non_neg_integer()
+
+  @doc """
+  Calculates the quarter of the year from the given `year`, `month`, and `day`.
+  """
+  @callback quarter_of_year(year, month, day) :: non_neg_integer()
+
+  @doc """
+  Calculates the year and era from the given `year`.
+  """
+  @callback year_of_era(year) :: {year, era :: non_neg_integer()}
```

Comment:
What do you think about defining the `era` type? It could be reused in the rest of the implementations